[0.8] test_query failure

Hi,

I tried to build the latest sunpy version for upload as Debian package, and got the following error during test:
```
__________________________________ test_query __________________________________

    @example(a.Time("2006-08-01", "2006-08-01"))
>   @example(a.Time("1983-05-01", "1983-05-02"))
    # This example tests a time range with a satellite jump and no overlap
    @example(a.Time("2009-11-30", "2009-12-3"))
    @given(goes_time())
    def test_query(time):

sunpy/net/dataretriever/tests/test_goes_ud.py:63: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python2.7/dist-packages/hypothesis/executors.py:58: in default_new_style_executor
    return function(data)
/usr/lib/python2.7/dist-packages/hypothesis/core.py:111: in run
    return test(*args, **kwargs)
sunpy/net/dataretriever/tests/test_goes_ud.py:73: in test_query
    qr1 = LCClient.search(time, Instrument('XRS'))
sunpy/net/dataretriever/client.py:314: in search
    self.map_.get('TimeRange'), **kwergs)
sunpy/net/dataretriever/sources/goes.py:87: in _get_url_for_timerange
    satellitenumber = kwargs.get('satellitenumber', self._get_goes_sat_num(date))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sunpy.net.dataretriever.sources.goes.XRSClient object at 0x7f93b86a6890>
date = datetime.datetime(1983, 5, 1, 0, 0)

    def _get_goes_sat_num(self, date):
        """[...]"""
        goes_operational = {
            2: TimeRange('1981-01-01', '1983-04-30'),
            5: TimeRange('1983-05-02', '1984-07-31'),
            6: TimeRange('1983-06-01', '1994-08-18'),
            7: TimeRange('1994-01-01', '1996-08-13'),
            8: TimeRange('1996-03-21', '2003-06-18'),
            9: TimeRange('1997-01-01', '1998-09-08'),
            10: TimeRange('1998-07-10', '2009-12-01'),
            11: TimeRange('2006-06-20', '2008-02-15'),
            12: TimeRange('2002-12-13', '2007-05-08'),
            13: TimeRange('2006-08-01', '2006-08-01'),
            14: TimeRange('2009-12-02', '2010-10-04'),
            15: TimeRange('2010-09-01', datetime.datetime.utcnow())
        }
    
        results = []
        for sat_num in goes_operational:
            if date in goes_operational[sat_num]:
                # if true then the satellite with sat_num is available
                results.append(sat_num)
    
        if results:
            # Return the newest satellite
            return max(results)
        else:
            # if no satellites were found then raise an exception
            raise ValueError('No operational GOES satellites on {}'.format(
>               date.strftime(TIME_FORMAT)))
E           ValueError: No operational GOES satellites on 1983-05-01 00:00:00

sunpy/net/dataretriever/sources/goes.py:55: ValueError
---------------------------------- Hypothesis ----------------------------------
Falsifying example: test_query(time=<Time(datetime.datetime(1983, 5, 1, 0, 0, 1), datetime.datetime(1983, 5, 1, 0, 0, 1), None)>)
```

--- a/sunpy/net/dataretriever/tests/test_goes_ud.py
+++ b/sunpy/net/dataretriever/tests/test_goes_ud.py
@@ -60,20 +60,22 @@ def test_fixed_satellite():
 
 
 @example(a.Time("2006-08-01", "2006-08-01"))
-@example(a.Time("1983-05-01", "1983-05-02"))
 # This example tests a time range with a satellite jump and no overlap
 @example(a.Time("2009-11-30", "2009-12-3"))
 @given(goes_time())
 def test_query(time):
-    tr = TimeRange(time.start, time.end)
-    if parse_time("1983-05-01") in tr:
+    qr1 = LCClient.search(time, Instrument('XRS'))
+    assert isinstance(qr1, QueryResponse)
+    assert qr1.time_range().start == time.start
+    assert qr1.time_range().end == time.end
+
+
+def test_query_error():
+    times = [a.Time("1983-05-01", "1983-05-02")]
+    for time in times:
         with pytest.raises(ValueError):
             LCClient.search(time, Instrument('XRS'))
-    else:
-        qr1 = LCClient.search(time, Instrument('XRS'))
-        assert isinstance(qr1, QueryResponse)
-        assert qr1.time_range().start == time.start
-        assert qr1.time_range().end == time.end
+
 
 @pytest.mark.skip(reason="Hangs with pytest only")
 @pytest.mark.online

